pkg/utils/overriding/merging.go

```diff
@@ -84,17 +90,21 @@ func mergeKeyedDevWorkspaceTemplateSpec(
 		return nil, err
 	}
 
-	result := keyedDevWorkspaceTemplateSpecContent{}
+	result := workspaces.DevWorkspaceTemplateSpecContent{}
 
 	preStartCommands := sets.String{}
 	postStartCommands := sets.String{}
 	preStopCommands := sets.String{}
 	postStopCommands := sets.String{}
 
 	for _, content := range allContents {
+
+		// TODO: Generate a function to implement this merging logic directly from
```

Comment:
Such TODOs in code are not well visible. It think it makes sense to register an issue for that instead of in addition.